### PR TITLE
Fixes issue #5804 - SQLLab - Saving a query with the same label resul…

### PR DESCRIPTION
**Pull request by fabiomendescom**
This fixes issue #5804.

### CODE REVIEW

**Comment by fabiomendescom** [comment]
Guys, anything else I need to do or can this be merged?

**Comment by fabiomendescom** [comment]
@kristw , this is related to the pull request  https://github.com/apache-superset/superset-ui/pull/105 you committed in superset-ui today. They depend on each other to work. So this should be ok to commit when you have time.

**Comment by kristw** [comment]
If you depend on the changes from `@superset-ui`, you have to wait a bit as we haven't bumped version of `@superset-ui` in this repo yet. There are a few small breaking changes that have to be address post-bump.

File: superset/assets/src/SqlLab/actions/sqlLab.js [hunk 1]
```diff
@@ -77,14 +77,27 @@ export function resetState() {
...
-  return dispatch =>
-    SupersetClient.post({
-      endpoint: '/savedqueryviewapi/api/create',
-      postPayload: query,
-      stringify: false,
-    })
-      .then(() => dispatch(addSuccessToast(t('Your query was saved'))))
-      .catch(() => dispatch(addDangerToast(t('Your query could not be saved'))));
+     let ret;
+     if (isNaN((query.id))) {
```
Comment (kristw) [inline superset/assets/src/SqlLab/actions/sqlLab.js]: `Number.isNaN()` instead of `isNaN()`.
Also remove the extra parentheses

**Comment by mistercrunch** [comment]
Let's get this through! Please fix the build issues

**Pull request closed without merging**